### ISSUE TITLE
"single pointer" definition not quite accurate?

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#749]
the (new in 2.1) definition of "single pointer" is, to my mind, not quite accurate.

> pointer input that operates with one point of contact with the screen, including single taps and clicks, double-taps and clicks, long presses, and path-based gestures

https://www.w3.org/TR/WCAG21/#dfn-single-pointer

noting that a pointer (at least in the Pointer Events sense) also includes mice, which don't have a "contact with the screen" (only in a very abstracted way)

a possible clarification here could be

> pointer input that operates with one point of contact with the screen / targets only one specific position on the page, including single taps and clicks, double-taps and clicks, long presses, and path-based gestures. This includes mouse inputs and interactions with a single finger or stylus with a touchscreen.

(or something to that effect)

happy to submit a PR if this sounds like something we'd want to clarify.

as pointer gesture and pointer cancellation SCs (which reference this key term) do apply to mouse already, changing this would likely be an editorial errata?

**Comment by jake-abma** [comment]
related to https://github.com/w3c/wcag/issues/746

Please comment on 2 & 3 about if path-based gestures needs to be here.

1. > The "one point of contact" can be read as: "only one point on the screen" not allowing hundreds of points when moving through a path
3. > The "one point of contact" can be read as: "one point at a time, but this (one) point may change hundreds of times due to moving the pointer

**Comment by patrickhlauke** [comment]
i believe the definition is not so much currently defining "single pointer", but "single pointer gesture" here (as it lists all the various interactions that counts as a "gesture", like taps/clicks, path-based movements, etc.

if just defining a "single pointer" per se, it should then probably just focus/emphasise that it stands in contrast to multi-pointer (really, multi-touch in practice, as most [possibly all] devices don't allow simulatenous use of different pointers like two mice, of a mouse and a single finger or stylus, or even multiple styluses).

so taking it further, if just trying to define "single pointer" (rather than single pointer gestures, which wouldn't be appropriate as the term is referenced from 2.5.2 as well), it probably needs a more fundamental rewrite...

> certain devices allow the use of more than one simultaneous pointer, such as multiple fingers on a touchscreen device. Single pointer interactions are exclusively limited to interactions where only one point of contact with the screen / only one specific position on the page is used. This includes mouse inputs and interactions with a single finger or stylus with a touchscreen.

**Comment by patrickhlauke** [comment]
maybe then expand that definition (rather than creating another one) to clarify/list

> pointer gestures include taps and clicks, double-taps and clicks, long presses, and path-based gestures. single pointer gestures only require a single pointer to be executed, while multi-pointer gestures (e.g. two-finger swipes and pinch-to-zoom interactions) involve two or more pointers used simultaneously.

**Comment by patrickhlauke** [comment]
note the definition of "pointer" in the Pointer Events specification

> A hardware agnostic representation of input devices that can target a specific coordinate (or set of coordinates) on a screen, such as a mouse, pen, or touch contact.

https://www.w3.org/TR/pointerevents/#dfn-pointer

This can possibly be leveraged for the definition here as well, prefixing the propsed bits above. so, combining the lot...

> **Single pointer**
> Pointer Events [REF] define a pointer as a hardware agnostic representation of input devices that can target a specific coordinate (or set of coordinates) on a screen, such as a mouse, pen, or touch contact. certain devices allow the use of more than one simultaneous pointer, such as multiple fingers on a touchscreen device. Single pointer interactions are exclusively limited to interactions where only one point of contact with the screen / only one specific position on the page is used. This includes mouse inputs and interactions with a single finger or stylus with a touchscreen. pointer gestures include taps and clicks, double-taps and clicks, long presses, and path-based gestures. Single pointer gestures only require a single pointer to be executed, while multi-pointer gestures (e.g. two-finger swipes and pinch-to-zoom interactions) involve two or more pointers used simultaneously.

**Comment by jake-abma** [comment]
This makes the case even bigger to extent the name of the definition, so if a pointer is a hardware representation (I also read it this way, that's why I created the issue) the name should become something like: "Single pointer interaction" or "Single pointer gesture"

**Comment by jake-abma** [comment]
also like the one from @detlevhfischer https://github.com/w3c/wcag/issues/746#issuecomment-494162692 'single point activation'

**Comment by patrickhlauke** [comment]
that would need changes to the two normative SCs that use that definition (though this would be a change that doesn't affect the meaning/applicability of the SCs)

> 2.5.1 Pointer Gestures
> All functionality that uses multipoint or path-based gestures for operation can be operated with a **single pointer** without a path-based gesture, unless a multipoint or path-based gesture is essential.
>
> 2.5.2 Pointer Cancellation
> For functionality that can be operated using a **single pointer**, at least one of the following is true

not sure if that's worth all the trouble?

**Comment by jake-abma** [comment]
will it be worth the trouble if we make it way more understandable for all people reading WCAG 2.1 (and lots of time NOT reading the definitions), or leave it as is and may end up with (maybe / at least) more than half of all people getting it wrong?

:-)

**Comment by patrickhlauke** [comment]
if we expand the definition (keeping it under the "single pointer" term) as per https://github.com/w3c/wcag/issues/749#issuecomment-494167564 but keep the normative wording in the SCs i think that would still work? just thinking it would meet less resistance from wider AGWG and related standards that have already copied 2.1's SCs wholesale and incorporated them.

**Comment by detlevhfischer** [comment]
I think a single pointer definition, clarifying the 'point of contact' issue to include indirect contact/cursor position, would be fine and can include single pointer paths, so these SCs could remain as the are.
We then still need a definition of 'path-based' capturing the decision about scope. I believe the exception from 2.1.1 might not be th ebest starting point because it is clearly not aimed at operational gestures, what counts as path-based, So the definition would need to clarify if 

- dragging is included
- only directional dragging (directional by perceived affordance or by actual freedom of pointer movement)
- directional drag and drop (dnd sortable lists)
- dragging for enlarging input elements by dragging corner etc. etc. 

@awkawk suggested to draw up a list of use cases to enable a wider deliberation of what should (or should not) be included in "path-based". (Is anyone picking up this action?)

**Comment by patrickhlauke** [comment]
was there any resolution on this particular aspect? happy to do a PR for just the definition wording itself, or the wider changes like "single pointer" to "single pointer interaction" in normative wordings as well...

**Comment by jake-abma** [comment]
we need an update for the definition but no official resolution so far. The wider change to "single pointer interaction" makes it much more clear what we mean but may encounter resistance. (We can change it for 2.2 more easily though with clear reasons and thus we might argue why not now)

**Comment by alastc** [comment]
Hi @patrickhlauke, I think an update to the definition to be clear about including mouse pointers is the best approach, a PR would be appreciated.

Maybe a bit cut down from your initial suggestion above, e.g:
> Single pointer
> A single pointer involve only one point of contact with the screen, or one specific position on the page. Interactions using s single pointer includes mouse input, or a single finger or stylus on a touchscreen. Single pointer operations include taps and clicks, double-taps and clicks, long presses, and dragging.
> 
> Note: Pointer Events [REF] define a pointer as a hardware agnostic representation of input devices that can target a specific coordinate (or set of coordinates) on a screen, such as a mouse, pen, or touch contact.

If it is really clear what a single pointer is, I don't think it needs comparing to multi-pointers.

**Comment by patrickhlauke** [comment]
@alastc done...with a bit more rewriting/tightening https://github.com/w3c/wcag/pull/809

**Comment by patrickhlauke** [comment]
a gentle nudge/ping on this and the related PR #809
